plugins/catalog-backend/src/service/request/parseEntityTransformParams.ts

```diff
@@ -38,24 +38,29 @@ function getPathArrayAndValue(input: Entity, field: string) {
 
 export function parseEntityTransformParams(
   params: Record<string, unknown>,
+  extra?: string[],
 ): ((entity: Entity) => Entity) | undefined {
-  const fieldsStrings = parseStringsParam(params.fields, 'fields');
-  if (!fieldsStrings) {
-    return undefined;
-  }
+  const queryFields = parseStringsParam(params.fields, 'fields');
 
-  const fields = fieldsStrings
-    .map(s => s.split(','))
-    .flat()
-    .map(s => s.trim())
-    .filter(Boolean);
+  const fields = Array.from(
+    new Set(
+      [...(extra ?? []), ...(queryFields ?? [])]
+        .map(s => s.split(','))
```

Comment:
Perhaps best to avoid splitting the `extra` since we have the more explicit input format in the post body?

Reply:
quite right, good catch - fix'd